packages/astro/src/types/public/config.ts

```diff
@@ -1699,9 +1699,46 @@ export interface ViteUserConfig extends OriginalViteUserConfig {
 		 * To use this feature with the Astro VS Code extension, you must also enable the `astro.content-intellisense` option in your VS Code settings. For editors using the Astro language server directly, pass the `contentIntellisense: true` initialization parameter to enable this feature.
 		 */
 		contentIntellisense?: boolean;
+
+		/**
+		 * @docs
+		 * @name experimental.responsiveImages
+		 * @type {boolean}
+		 * @default `undefined`
+		 * @version 5.0.0
+		 * @description
+		 *
+		 * Enables and configures automatic responsive image generation for images in your project. Set to `true` or an object with configuration options.
+		 * When enabled it will not transform any images unless you pass the `layout` option to the image component, or set the `layout` option in the `responsiveImages` configuration.
+		 *
+		 * ```js
+		 * {
+		 *  experimental: {
+		 * 		responsiveImages: {
+		 * 			layout: 'responsive',
+		 * 		},
+		 * }
+		 * ```
+		 *
+		 */
+
+		responsiveImages?:
```

Comment:
You mean as `experimental.image`, or at the top level?

Reply:
Yeah, the normal top level `image`

Reply:
Is that something we do for experimental features?

Reply:
We kinda have two philosophies, some people put it under experimental, some others don't. I personally think it makes the upgrading smoother when it's already where it will be, as you then don't need to move anything, but it's up to you really. It just felt a bit jarring to have both `image` and `responsiveImages` as two different options!

`image` in fact was top level while experimental, for instance

Reply:
The main reason I didn't do that is because it would mean there were options showing up as available in the `image` config that don't work unless the experimental feature is enabled. I do wonder if maybe it should be `experimental.image`, or would that be even more confusing!

Reply:
That works perfectly for me!

Reply:
@sarah11918 what are your thoughts on this? Should the config options for the feature go inside `experimental.responsiveImages`, or shoudl that just be a boolean, with the config sitting in the top-level `image` object? I'm conflicted!

Reply:
I saw this discussion and was happy to let it play out. :sweat_smile: 

We have precedence for both! `experimental.env` kept all its config within the experimental config and there was a time where `astro:i18n` had an option outside of the experimental config.

I don't have a strong opinion -- people re going to have to update their config either way in the cases you've described: either they move `layout` to the `image` option directly OR they update `image.experimentalLayout` to just `layout`. It *sort of* to me makes more sense in that case to keep all the config together?

The other option is also a bit wonky: just use `image.layout` BUT it only does anything with the experimental flag set. THIS is the option that means people don't have to update their config when it comes out of experimental (other than remove the flag). It's also not necessarily pretty to document something in the main config that is essentially no op without the flag, and, it means sending the user to two different places in order to configure the experimental feature. BUT it's the only thing that has the true advantage of needing less updating on behalf of the user.

So, I won't fight any of these options. I can make them all work and there's no clear and obvious winner here. I won't make the final call unless someone specifically wants me to.

Reply:
In the absence of any strong opinions on either side, I'll leave it as-is

Reply:
Changed my mind again! I've gone for the second option